services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3CrtResponseHandlerAdapter.java

```diff
@@ -86,8 +108,7 @@ public int onResponseBody(ByteBuffer bodyBytesIn, long objectRangeStart, long ob
                 failResponseHandlerAndFuture(failure);
```

Comment:
If the write future failed, we only close on the meta request, which won't stop the meta request keep receiving data, can we also cancel on the meta request before it closes? 

The only result of cancel on the meta request will be the complete callback may invoked with `cancelled` error, but as the result has already been set, it probably doesn't affect anything.

Reply:
Yeah, makes sense, I'll just call cancel whenever there's an exception